### PR TITLE
FEATURE: Add `in:polls` filter to search

### PR DESCRIPTION
**Pull request by pmusaraj**
Allows users to filter search results to posts with polls (if that plugin is enabled).

### CODE REVIEW

File: plugins/poll/spec/lib/search_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  before do
+    SearchIndexer.enable
+    Jobs.run_immediately!
+
+    SearchIndexer.index(topic2, force: true)
+    SearchIndexer.index(topic, force: true)
+  end
+
+  after do
+    SiteSetting.poll_enabled = true
```
Comment (jjaffeux) [inline plugins/poll/spec/lib/search_spec.rb:27]: is this needed?

File: plugins/poll/spec/lib/search_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    SearchIndexer.disable
+  end
+
+  describe "in:polls" do
+    it "allows filtering search to posts containing polls" do
+      results = Search.execute("rainbow in:polls", guardian: Guardian.new())
```
Comment (jjaffeux) [inline plugins/poll/spec/lib/search_spec.rb:33]:
```suggestion
      results = Search.execute("rainbow in:polls", guardian: Guardian.new)
```

File: plugins/poll/spec/lib/search_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      expect(results.posts).to contain_exactly(post_with_poll)
+    end
+
+    it "ignores in:polls if plugin is disabled" do
+      SiteSetting.poll_enabled = false
+      results = Search.execute("rainbow in:polls", guardian: Guardian.new())
```
Comment (jjaffeux) [inline plugins/poll/spec/lib/search_spec.rb:39]:
```suggestion
      results = Search.execute("rainbow in:polls", guardian: Guardian.new)
```

File: plugins/poll/spec/lib/search_spec.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    SearchIndexer.index(topic, force: true)
+  end
+
+  after do
+    SiteSetting.poll_enabled = true
+    SearchIndexer.disable
+  end
+
+  describe "in:polls" do
+    it "allows filtering search to posts containing polls" do
```
Comment (jjaffeux) [inline plugins/poll/spec/lib/search_spec.rb:32]:
```suggestion
  context "when using in:polls" do
    it "filters posts containing polls" do
```

Feel free to ignore but I think it reads better.

File: plugins/poll/spec/lib/search_spec.rb [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      results = Search.execute("rainbow in:polls", guardian: Guardian.new())
+      expect(results.posts).to contain_exactly(post_with_poll)
+    end
+
+    it "ignores in:polls if plugin is disabled" do
```
Comment (jjaffeux) [inline plugins/poll/spec/lib/search_spec.rb:37]: Smililary here I would do:

```
context "when polls are disabled" do
  it "ignores in:polls filter" do
    # ...
  end
end
```

**Comment by jjaffeux** [comment]
LGTM :+1: only nitpicks and minor questions/suggestions

**Comment by pmusaraj** [comment]
Thanks for the review @jjaffeux, I integrated the suggestions.

**Pull request merged**